README.md

```diff
@@ -180,6 +180,43 @@ If you have issues building ruby-libvirt, try the following:
 ```shell
 CONFIGURE_ARGS='with-ldflags=-L/opt/vagrant/embedded/lib with-libvirt-include=/usr/include/libvirt with-libvirt-lib=/usr/lib' GEM_HOME=~/.vagrant.d/gems GEM_PATH=$GEM_HOME:/opt/vagrant/embedded/gems PATH=/opt/vagrant/embedded/bin:$PATH vagrant plugin install vagrant-libvirt
 ```
+### Additional Notes for Fedora and Similar Linux Distributions
+
+If you encounter the following load error when using the vagrant-libvirt plugin (note the required by libssh):
+
+```shell
+/opt/vagrant/embedded/lib/ruby/2.4.0/rubygems/core_ext/kernel_require.rb:55:in `require': /opt/vagrant/embedded/lib64/libcrypto.so.1.1: version `OPENSSL_1_1_1b' not found (required by /lib64/libssh.so.4) - /home/xxx/.vagrant.d/gems/2.4.6/gems/ruby-libvirt-0.7.1/lib/_libvirt.so (LoadError)
+```
+then the following steps have been found to resolve the problem. Thanks to James Reynolds (see https://github.com/hashicorp/vagrant/issues/11020#issuecomment-540043472). The specific version of libssh will change over time so references to the rpm in the commands below will need to be adjusted accordingly.
+
+```shell
+dnf download --source libssh
+rpm2cpio libssh-0.9.0-5.fc30.src.rpm | cpio -imdV
+tar xf libssh-0.9.0.tar.xz
+mkdir build
+cd build
+cmake ../libssh-0.9.0 -DOPENSSL_ROOT_DIR=/opt/vagrant/embedded/
+make
+sudo cp lib/libssh* /opt/vagrant/embedded/lib64
+```
+
+If you encounter the following load error when using the vagrant-libvirt plugin (note the required by libk5crypto):
+
+```shell
+/opt/vagrant/embedded/lib/ruby/2.4.0/rubygems/core_ext/kernel_require.rb:55:in `require': /usr/lib64/libk5crypto.so.3: undefined symbol: EVP_KDF_ctrl, version OPENSSL_1_1_1b - /home/rbelgrave/.vagrant.d/gems/2.4.9/gems/ruby-libvirt-0.7.1/lib/_libvirt.so (LoadError)
+```
+
+then the following steps have been found to resolve the problem. After the steps below are complete, then reinstall the vagrant-libvirt plugin. Thanks to Marco Bevc (see https://github.com/hashicorp/vagrant/issues/11020#issuecomment-625801983):
+
+```shell
+dnf download --source krb5-libs
+rpm2cpio krb5-1.18-1.fc32.src.rpm | cpio -imdV
+tar xf krb5-1.18.tar.gz
+cd krb5-1.18/src
+./configure
+make
+sudo cp -a lib/crypto/libk5crypto.* /opt/vagrant/embedded/lib64/
```

Comment:
Is there any reason you couldn't just copy the file from the current system into the directory without needing to compile again from source? The previous item it make sense because it is linking against some libraries provided by the vagrant upstream package, but for this one?

Reply:
The libssh build explicitly uses the openssl library packaged with upstream vagrant to resolve the missing symbol problem. For the kerberos package, it would seem feasible to just copy the libk5crypto.so* files from /usr/lib64 to the /opt/vagrant/embedded/lib64 directory and then doing the plugin installation. I tried that at one point and it did not work (for reasons that elude me still).  Marco Bevc provided the build steps that I copied - I would have to defer to input from him. I should probably create a clean vm with fedora 32 workstation and try a few alternatives to try and answer the question.

Reply:
It should be possible to confirm locally, if you copy across the file from /usr/lib64 to /opt/vagrant/embedded/lib64/ (ensuring it replaces the version currently there), either you can vagrant up successfully which means that you don't need to compile, or you'll get an error that indicates the system library has additional requirements and by compiling from source you skip pulling those in which is why the instructions above are needed. Shouldn't need a pristine fedora 32 system to verify

Reply:
I think I know why just using the system libraries doesn't work. The recompile here skips linking with certain other libraries, which avoids the issue of the missing symbols. There are issues trying to get this to work for CentOS however I'll try to verify on Fedora and merge

Reply:
Confirmed, this works as needed on Fedora as the complied krb5 libraries aren't linked against openssl, the spec file for the packge contains the following done during the prep stage:
```
    --with-crypto-impl=openssl \
    --with-tls-impl=openssl \
```

That would also explain why can't use `rpmbuild -bp <specfile>` to set up a build tree for it.